### ISSUE TITLE
[BUG] - DB sync after upgrading to 1.26.1 => cardano-cli: MuxIOException writev: resource vanished 

### ISSUE THREAD

**Issue by robinboening** [input-output-hk/cardano-node#2597]
*External* 

**Area**
Syncing the DB after upgrading node to v1.26.1

**Summary**
I upgraded a cardano-node from v1.25.1 to v1.26.1. The legacy DB wasn't fully synced with the network (~94%) when I started the node. The node correctly started an immediate DB migration (roughly 90min with my 3,1 GHz Intel Core i5) and succeeded. Afterwards it started the usual sync but after a few minutes (at around 95%) it errored out with the following message

```
MuxError (MuxIOException writev: resource vanished (Broken pipe)) "(sendAll errored)"
cardano-cli: MuxIOException writev: resource vanished (Broken pipe) "(sendAll errored)"(standard_in) 2: syntax error
``` 

I then tried again. This time I got this error

```
cardano-cli: MuxIOException Network.Socket.recvBuf: resource vanished (Connection reset by peer) "(recv errored)"(standard_in) 2: syntax error
```

**Expected behavior**
I expect the DB to sync without errors.

**System info (please complete the following information):**
- OS: Debian
- Version: Buster (v10.8)
- Node version: 1.26.1

**Comment by ivantomac** [comment]
I was getting that too. After adding `+RTS -M3G -RTS` as parameters to cardano-node, which limits the maximum heap size to 3 gigabytes, the problem went away. I am guessing 1.26.1 by default uses more RAM than the older versions and is running out of memory.

See https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/runtime_control.html for more info on tweaking runtime parameters of GHC compiled binaries.

**Comment by robinboening** [comment]
For me it worked after pulling a "fully" synced db from a live node.
I could close this issue as its working for me now, but I believe the actual problem is not solved, I just found a workaround. So I'll leave it open for now but feel free to close it whenever you want.

**Comment by frog357** [comment]
I am seeing this issue multiple times a day on my mint tool after upgrading from 1.29. It only seems to happen about 1 in 100 transactions. We have 16GB ram total - only running a bash script and Cardano-node, nothing else is active on this host.
We only have slightly more than 4GB of free ram and this workstation is not offering any public services other than Cardano-node.

MuxError (MuxIOException writev: resource vanished (Broken pipe)) “(sendAll errored)”
cardano-cli: MuxIOException Network.Socket.recvBuf: resource vanished

**Comment by zhaozhijun1988** [comment]
I am seeing this issue ,how todo?

**Comment by johnalotoski** [comment]
Also seeing this same issue on cardano-node `1.30.1` with cardano-cli.  On average, perhaps once in 20 or 30 tip queries for example, but it can vary widely.  Replication of the error can be seen quickly with something like the following (add a `sleep 1` to the loop below if wanted):

`while true; cardano-cli query tip <--mainnet|--testnet-magic X>; done`

Output from the above intermittently shows some queries return with the additional error the same as @frog357's above:
```
cardano-cli: MuxIOException writev: resource vanished (Broken pipe) "(sendAll errored)"
MuxError (MuxIOException writev: resource vanished (Broken pipe)) "(sendAll errored)"
```

I do have another system also running `1.30.1` which this doesn't happen on.  Not sure what the difference would be.  Both systems have plenty of RAM available.

**Comment by johnalotoski** [comment]
This appears to be resolved in 1.31.0.  See: https://github.com/input-output-hk/cardano-node/issues/3042#issuecomment-960553159.  Testing confirms the error no longer appears for me on 1.31.0

**Comment by Jimbo4350** [comment]
Closing this. If this is still relevant please reopen.
